tests/test_fields.py

```
@@ -630,3 +633,172 @@ def x(self) -> int:
+    item = await page.to_item()
+    assert item == BigItem(x=1, y=2, z=None)
+    assert page.fields_to_extract == ["x", "y"]
+    assert page.call_counter == {"x": 1, "y": 1}
+
+    # Select all fields
+    page = BigPage(SelectFields(fields={"*": True}))
+    item = await page.to_item()
+    assert item == BigItem(x=1, y=2, z=3)
+    assert page.fields_to_extract == ["x", "y", "z"]
+    assert page.call_counter == {"x": 1, "y": 1, "z": 1}
+
+    # Don't select all fields
+    page = BigPage(SelectFields(fields={"*": False}))
+    with pytest.raises(TypeError, match=expected_type_error_msg):
+        await page.to_item()
+    assert page.fields_to_extract == []
+    assert page.call_counter == {}
+
+    # Exclude all but one
+    page = BigPage(SelectFields(fields={"*": False, "x": True}))
+    item = await page.to_item()
+    assert item == BigItem(x=1, y=None, z=None)
+    assert page.fields_to_extract == ["x"]
+    assert page.call_counter == {"x": 1}
+
+    # Include all fields but one
+    page = BigPage(SelectFields(fields={"*": True, "z": False}))
+    item = await page.to_item()
+    assert item == BigItem(x=1, y=2, z=None)
+    assert page.fields_to_extract == ["x", "y"]
+    assert page.call_counter == {"x": 1, "y": 1}
+
+    # overlapping directives on the same field should be okay
+    page = BigPage(SelectFields(fields={"*": True, "x": True, "y": True, "z": True}))
+    item = await page.to_item()
+    assert item == BigItem(x=1, y=2, z=3)
+    assert page.fields_to_extract == ["x", "y", "z"]
+    assert page.call_counter == {"x": 1, "y": 1, "z": 1}
+
+    # Excluding a required field throws an error
+    page = BigPage(SelectFields(fields={"x": False}))
+    with pytest.raises(TypeError, match=expected_type_error_msg):
+        item = await page.to_item()
+    assert page.fields_to_extract == ["y"]
+    assert page.call_counter == {"y": 1}
+
+    # boolean-like values are not supported. They are simply ignored and the
```

I think we should raise an error, not ignore wrong values.